pkg/api/plugin.go

```diff
@@ -14,6 +14,34 @@ const (
 	ContextKeyTenantID     ContextKey = "TenantID"
 )
 
+type PluginStep string
+
+const (
+	PluginStepDeploy                  PluginStep = "Deploy"
+	PluginStepInitialize              PluginStep = "InitializeCluster"
+	PluginStepUpdateDrain             PluginStep = "UpdateDrain"
+	PluginStepUpdateMasterVMRotation  PluginStep = "UpdateMasterVMRotation"
+	PluginStepUpdateInfraVMRotation   PluginStep = "UpdateInfraVMRotation"
+	PluginStepUpdateComputeVMRotation PluginStep = "UpdateComputeVMRotation"
+	PluginStepWaitForHealthz          PluginStep = "WaitForHealthz"
+	PluginStepWaitForNodes            PluginStep = "WaitForNodes"
+	PluginStepHealthCheck             PluginStep = "HealthCheck"
+	PluginStepWaitForInfraDaemonSets  PluginStep = "WaitForInfraDaemonSets"
```

Comment:
Not sure it's worth separating between DaemonSets and Deployments. `WaitForInfraServices` is probably more appropriate.

Reply:
see: https://github.com/openshift/openshift-azure/issues/672#issuecomment-434546004